### DISCUSSION TITLE
Mapping a GitHub App id into the App's name

### DISCUSSION THREAD

**Discussion by hwine** [community/community#24821] [Apps, API and Webhooks] [answered]
<p>I want to get a list of the GitHub applications being used by my organization. The <a href="https://developer.github.com/changes/2019-10-23-get-installations-for-org/" rel="nofollow noopener">newish v3 API call</a> returns that list, but only identifies the application by its id. I want to convert the id into the name of the app.</p>
<p>I can’t find an API I can use to translate that id into a slug, which is required to obtain more information about the app, per the <a href="https://developer.github.com/v3/apps/#get-a-single-github-app" rel="nofollow noopener">Get a Single GitHub App</a>, which also appears to be the only endpoint that can be used without authenticating as an app.</p>
<p>While I would prefer a v3 solution, a v4 would be fine as well.</p>

**UPDATE:** the v3 API version 2022-11-28 [list app installations for an organization](https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#list-app-installations-for-an-organization) response does return the `app_slug`.
_[this question was originally asked during the beta period]_

**Comment by hwine** [comment]
<p>No solution found, filed an enhancement request to have “:slug” added to the response.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/hwine">@hwine</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list.</p>
<p>Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Reply by hwine** [reply, marked as answer]
**UPDATE:** the v3 API version 2022-11-28 [list app installations for an organization](https://docs.github.com/en/rest/orgs/orgs?apiVersion=2022-11-28#list-app-installations-for-an-organization) response does return the `app_slug`.
_[this question was originally asked during the beta period]_

**Comment by maboloshi** [comment]
[Get an organization installation for the authenticated app](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-an-organization-installation-for-the-authenticated-app)

The "app_slug" in the response is probably what you need.
However, you need to build the JWT yourself before making the request.
[Generate a JWT](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app)

ps:
[Get a repository installation for the authenticated app](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-a-repository-installation-for-the-authenticated-app)
[Get a user installation for the authenticated app](https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-a-user-installation-for-the-authenticated-app)

**Reply by hwine** [reply]
Thanks for your reply, but the calls you cite only work for an _authenticated_ app. My use case is discovering apps-in-use without being authenticated to the apps.
